tests/Tensor_test_2.cpp

```diff
@@ -0,0 +1,762 @@
+  A5.Add_(B5);
+  ValuesList V5{2,2};
+  CheckTensStruct(A5, Type.Double, Device.cpu, {1,2});
+  tens2test(A5, V5);
+}
+
+TEST(Tensor, product){
+  // *
+  auto A1 = arange(12).reshape(3,4);
+  auto B1 = ones({3,4})*4;
+  auto C1 = A1*B1;
+  ValuesList V1{0,4,8,12,16,20,24,28,32,36,40,44};
+  CheckTensStruct(C1, Type.Double, Device.cpu, {3,4});
+  tens2test(C1, V1);
+
+  // *=
+  auto A2 = arange(12).reshape(3,4);
+  auto B2 = ones({3,4})*4;
+  A2 *= B2;
+  ValuesList V2{0,4,8,12,16,20,24,28,32,36,40,44};
+  CheckTensStruct(A2, Type.Double, Device.cpu, {3,4});
+  tens2test(A2, V2);
+
+  // MUL
+  auto A3 = arange(12).reshape(3,4);
+  auto B3 = ones({3,4})*4;
+  auto C3 = A3.Mul(B1);
+  ValuesList V3{0,4,8,12,16,20,24,28,32,36,40,44};
+  CheckTensStruct(C3, Type.Double, Device.cpu, {3,4});
+  tens2test(C3, V3);
+
+  // MUL_
+  auto A4 = arange(12).reshape(3,4);
+  auto B4 = ones({3,4})*4;
+  A4.Mul_(B4);
+  ValuesList V4{0,4,8,12,16,20,24,28,32,36,40,44};
+  CheckTensStruct(A4, Type.Double, Device.cpu, {3,4});
+  tens2test(A4, V4); 
+}
+
+TEST(Tensor, equal){
+  // ==
+  auto A1 = arange(6).reshape({2,3});
+  auto B1 = arange(6).reshape({2,3});
+  auto C1 = A1 + 1;
+  auto cprT = A1 == B1;
+  auto cprF = A1 == C1;
+  for (int i = 0; i < 2; i++){
+    for (int j = 0; j < 3; j++){
+      EXPECT_TRUE(cprT(i,j).item<bool>());
+      EXPECT_FALSE(cprF(i,j).item<bool>());
+    }
+  }
+}
+
+TEST(Tensor, cpr){
+  // cpr
+  auto A = arange(6).reshape({2,3});
+  auto B = arange(6).reshape({2,3});
+  auto C = A + 1;
+  auto cprT = A.Cpr(B);
+  auto cprF = A.Cpr(C);
+  for (int i = 0; i < 2; i++){
+    for (int j = 0; j < 3; j++){
+      EXPECT_TRUE(cprT(i,j).item<bool>());
+      EXPECT_FALSE(cprF(i,j).item<bool>());
+    }
+  }
+}
+
+
+TEST(Tensor, append){
+  // rank 1
+  auto A1 = ones({2});
+  A1.append(4);
+  A1.reshape_({1,3});
+  ValuesList V1{1,1,4};
+  CheckTensStruct(A1, Type.Double, Device.cpu, {1,3});
+  tens2test(A1, V1);
+
+  // multi-rank
+  auto A2 = ones({3,2});
+  auto B2 = ones({2})*3;
+  ValuesList V2{1,1,1,1,1,1,3,3};
+  A2.append(B2);
+  CheckTensStruct(A2, Type.Double, Device.cpu, {4,2});
+  tens2test(A2, V2);
+}
+
+TEST(Tensor, rank) {
+  // rank 2 tensor
+  for(auto& assign_device : device_list) {
+      for(auto& assign_dtype : dtype_list) {
+        Tensor T1 = ones({2, 2}, assign_dtype, assign_device);
+        EXPECT_EQ(T1.rank(), 2);
+      }
+  }  
+
+  // rank 3 tensor
+  for(auto& assign_device : device_list) {
+      for(auto& assign_dtype : dtype_list) {
+        Tensor T2 = ones({3, 3}, assign_dtype, assign_device);
+        EXPECT_EQ(T2.rank(), 2);
+      }
+  }
+}
+
+TEST(Tensor, SVD){
+  auto T1 = Tensor({2,2});
+  double values[2][2] = 
+  {
+    {1,2},
+    {1,2}
+  };
+  for (int i = 0; i < 2; i++){
+    for (int j = 0; j < 2; j++){
+      T1(i,j) = values[i][j];
+    }
+  }
+  auto SVDs = T1.Svd();
+  std::cout << SVDs[0] << std::endl;
+  std::cout << SVDs[1] << std::endl;
+  std::cout << SVDs[2] << std::endl;
+}
```

Comment:
This test need to have checking on the elements before/after SVD, as well as element of U, vT